### ISSUE TITLE
partial auc for pr_auc

### ISSUE THREAD

**Issue by kkmann** [tidymodels/yardstick#179]
Hi, 

I just find myself implementing a custom metric for a partial AUC of the PR curve. This seems to work already for AUC but not PR-AUC. 

Looking closer, yardstick computes the AUC using a simple internal trapezoid implementation 
https://github.com/tidymodels/yardstick/blob/46fad5115e03449174de0813710d9d4d50b4bd0c/R/prob-helpers.R#L6-L31

That is then called to compute pr_auc here
https://github.com/tidymodels/yardstick/blob/46fad5115e03449174de0813710d9d4d50b4bd0c/R/prob-pr_auc.R#L118

Would you be willing to merge a PR that allows for optional passing of lower and/or upper boundary for the AUC computation? I guess the best way to handle normalization is to renormalize by the width of the actual integration range?

**Comment by DavisVaughan** [comment]
What is the reasoning for a partial PR-AUC? I have seen partial AUCs computed for ROC-AUC, but never for PR-AUC. I can't seem to pull up any references where this is done. It would be very helpful if you had a reference!

If implemented, it would likely be done like sklearn's partial roc-auc. They have a `max_fpr` value that can be set. The equivalent here is probably `max_recall`, but I'm not sure of its practical use.
https://github.com/scikit-learn/scikit-learn/pull/3840/files

**Comment by kkmann** [comment]
Well, the PR-AUC essentially measures average performance over a range of sensitivity/recall values. Now, it might very well be that my application requires a certain minimum sensitivity, so I am not interested in the contribution of the PR-curve for smaller sensitivities, i.e., I should optimize partial PR-AUC. Partial PR-AUC is not inflated by performance improvments for practically irrelevant sensitivity values.

**Comment by juliasilge** [comment]
I have not seen this in practice really but FWIW looks like the [precrec](https://takayasaito.github.io/precrec/reference/part.html) package supports partial AUCs for both ROC and precision-recall curves.

**Comment by kkmann** [comment]
ah, thanks, nice + proper interpolation for PRC AUC!

**Comment by DavisVaughan** [comment]
At the current moment, we don't see much precedent for this method (beyond precrec). I'm going to close this one, but feel free to put in a PR if you really want it!
